File: python/src/NormalItem.py
from src.Item import Item
from src.updateable import updateable

class NormalItem(Item, updateable):
    def __init__(self, name, sell_in, quality):
        Item.__init__(self, name, sell_in, quality)
    
    def getSell_in(self):
        return self.sell_in
    
    def getQuality(self):
        return self.quality
    
    def updateQuality(self):
        if self.sell_in <= 0:
            self.quality -= 2
            self.sell_in -= 1
        else:
            self.quality -= 1
            self.sell_in -= 1
        if self.quality <= 0:
            self.quality = 0

File: java/src/main/java/edu/poniperro/gildedrose/NormalItem.java
package edu.poniperro.gildedrose;

public class NormalItem extends Item implements Updateable {

    public NormalItem(String name, int sell_in, int quality) {
        super(name, sell_in, quality);
    }

    @Override
    public void updateQuality() {
        if (getSell_in() <= 0) {
            setQuality(getQuality() - 2);
        } else {
            setQuality(getQuality() - 1);
        }
        setSell_in();
        checkQuality();
    }

    private void checkQuality() {
        if (getQuality() <= 0) {
            setQuality(0);
        }
    }
}